src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -316,5 +382,18 @@ private SemanticRange DataToSemanticRange(int lineDelta, int charDelta, int leng
             // We don't currently have any need for tokenModifiers
             yield return currentRange.Modifier;
         }
+
+        private async Task<(DocumentSnapshot Snapshot, int? Version)> TryGetDocumentInfoAsync(string absolutePath, CancellationToken cancellationToken)
+        {
+            var document = await Task.Factory.StartNew(() =>
```

Comment:
You don't capture the version here meaning it's always `null`

Reply:
I named the variable poorly, this is a tuple of `(DocumentSnapshot, int?)`, not just the snapshot.